### PR TITLE
add noop, and rename package

### PR DESCRIPTION
**Pull request by oibe**
- rename packages to be io.opentracing
- add noop Tracer, Span, and SpanBuilder

### CODE REVIEW

**Comment by yurishkuro** [comment]
LGTM

@bensigelman @michaelsembwever

**Comment by bensigelman** [comment]
Sorry, I thought I saw this get merged but was apparently incorrect!

@michaelsembwever I assume this is uncontroversial...

File: opentracing/src/main/java/io/opentracing/noop/NoopSpan.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.noop;
+
+import io.opentracing.Span;
+
+public class NoopSpan implements Span {
```
Comment (michaelsembwever) [inline opentracing/src/main/java/io/opentracing/noop/NoopSpan.java]: [trivial] why is this public? and why isn't it final?

File: opentracing/src/main/java/io/opentracing/noop/NoopSpanBuilder.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.noop;
+
+
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+
+public class NoopSpanBuilder implements Tracer.SpanBuilder {
```
Comment (michaelsembwever) [inline opentracing/src/main/java/io/opentracing/noop/NoopSpanBuilder.java]: [trivial] same question. why public? why not final?

File: opentracing/src/main/java/io/opentracing/noop/NoopSpan.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    private NoopSpan defaultSpan = new NoopSpan();
```
Comment (michaelsembwever) [inline opentracing/src/main/java/io/opentracing/noop/NoopSpan.java]: why not make this `static final` ?

File: opentracing/src/main/java/io/opentracing/noop/NoopSpanBuilder.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    private Tracer.SpanBuilder defaultSpanBuilder = new NoopSpanBuilder();
+    private Span defaultSpan = new NoopSpan();
```
Comment (michaelsembwever) [inline opentracing/src/main/java/io/opentracing/noop/NoopSpanBuilder.java]: It is odd that there's two NoopSpan instances created in the JVM. 

A method on the NoopSpan will return a different instance to that returned from the builder.
I'd rather see NoopSpan.DEFAULT_SPAN make package-protected, and it used in the builder here.

File: opentracing/src/main/java/io/opentracing/noop/NoopTracer.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ * the License.
+ */
+package io.opentracing.noop;
+
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+
+public class NoopTracer implements Tracer {
+
+    private SpanBuilder defaultSpanBuilder = new NoopSpanBuilder();
```
Comment (michaelsembwever) [inline opentracing/src/main/java/io/opentracing/noop/NoopTracer.java]: again why not `static final` ?

File: opentracing/src/main/java/io/opentracing/noop/NoopTracer.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.noop;
+
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+
+public class NoopTracer implements Tracer {
```
Comment (michaelsembwever) [inline opentracing/src/main/java/io/opentracing/noop/NoopTracer.java]: should be a `final` class.

**Comment by yurishkuro** [comment]
@michaelsembwever good points all around. 

@oibe let's address them.

**Pull request merged**